.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "deps/finch-core"]
```

Comment:
General question: in this case, do we still need to have Core release or just make Finch update to Core's latest commit in the dependency sync?

Reply:
We can actually have the submodule reference any commit on the finch-core repo, without having to cut a release. The process is a little different now, though.

Instead of changing a `CORE_URL` in our Makefile, you just have to:
1. checkout the repo (with submodules, instructions updated in CONTRIBUTING.md)
    1. If you already have the repo checked out locally, you can just run `git submodule update --init --recursive` instead of re-cloning
1. `cd deps/finch-core`
1. `git checkout <any commit / tag / branch>`, the commit does not have to be tied to a specific release
    1. This will update the reference in the `deps/finch-core` "file" that you see in the diff (e.g. in this diff: [deps/finch-core](https://github.com/runfinch/finch/pull/188/files#diff-4a2d9142926c1a8dcfa9ae228049330bb794487e6b38e0807a6cd1736aec28be))

Reply:
Should we track the latest commit in `main`  and create a workflow to update the submodule at a weekly cadence? Maybe even have an additional  trigger manually pull in latest changes from main?

Reply:
> Should we track the latest commit in `main` and create a workflow to update the submodule at a weekly cadence? Maybe even have an additional trigger manually pull in latest changes from main?
> 
> If we did this maybe we don't need to release `finch-core` anymore?

I think we can definitely do that going forward, but this PR just uses a fixed version since that's what we were already doing with `CORE_URL`.

Are you ok with adding the sync job in a separate PR?

Reply:
Yeah, not required in this PR. Just mentioned it as an enhancement in the future.